cmd/fs-v1-helpers.go

```diff
@@ -102,10 +102,25 @@ func fsMkdirAll(dirPath string) (err error) {
 		return errors.Trace(err)
 	}
 
-	if err = os.MkdirAll(dirPath, 0777); err != nil {
+	i := 0
+retry:
+	if err = os.MkdirAll(dirPath, mode); err != nil {
+		// File path cannot be verified since one of the parents is a file.
+		if isSysErrNotDir(err) {
+			return errFileAccessDenied
+		} else if isSysErrPathNotFound(err) {
+			// This is a special case should be handled only for
+			// windows, because windows API does not return "not a
+			// directory" error message. Handle this specifically here.
+			return errFileAccessDenied
+		}
+		// Retry only for the first no such file or directory error.
+		if os.IsNotExist(err) && i == 0 {
```

Comment:
Why only one retry ? I guess it is safe and better to increase it to 3.

Reply:
That is debatable @vadmeste - I would just keep it as is in this PR. 3 is too much IMO. Retrying it once is fine..